backend/video/views.py

```diff
@@ -98,13 +99,15 @@ def get_video(request):
             )
 
         url = f"https://drive.google.com/uc?export=download&confirm=yTib&id={file_id['data']}"
-        print("FILE ID", file_id["data"])
+
         # Get the video metadata
         title = urllib.request.urlopen(urllib.request.Request(url)).info().get_filename()
         direct_audio_url = url
-
-        # Set duration to 0
-        duration = timedelta(seconds=0)
+    
+        # Calculate the duration 
+        filename, headers = urllib.request.urlretrieve(url)
+        audio = MP3(filename)
```

Comment:
Just curious how it works - Is it actually a `filename`? Or `file` ?
If it's `file`, can we please rename to `file` to make it clear

Reply:
It is actually the filename - https://docs.python.org/3/library/urllib.request.html#urllib.request.urlretrieve

Reply:
Ohh OK.. One more question then:

```py
title = urllib.request.urlopen(urllib.request.Request(url)).info().get_filename()
```

Does this also download the full file just to get the filename?
Or does it manage to internally just read it using the HTTP header without having to download the file content?

Reply:
This doesn't download the full file. It uses the headers and metadata to read the filename